### DISCUSSION TITLE
Best practices for FEAT_DYNAMIC_REAL in DeepAR

### DISCUSSION THREAD

**Discussion by vikasmalhotra08** [awslabs/gluonts#1705] [Q&A] [answered]
Hello,

I have a question on FEAT_DYNAMIC_REAL. As per documentation and comments by folks including committers and my experience as well, we would need to have a known future values for any feature we use as part of the said field name. 

My questions are: 
- Assuming we just impute the mean values for certain features and concatenate it with other features like holidays and use it in FEAT_DYNAMIC_REAL, does this sound like a good strategy? If not, what are the downfalls of using this strategy.
- How do people handle this use-case in production settings? Is imputing mean values fair strategy or people have other strategies when using DeepAR? I am not the only one who has come across this, so I am assuming people have figured out ways to handle use-cases where covariates are not known in future. 

I really wished there was a Wiki on this repo where we could collate these best practices, it would definitely help out other folks.

**Comment by mbohlkeschneider** [comment, marked as answer]
Hi @vikasmalhotra08,

Thank you for your interest in GluonTS. 

Imputing future `feat_dynamic_real` with mean values does not work very well in my experience. I think it is plausible that it works if you have a feature that does not change much in the prediction range. 

The mean reason why there isn't a document that shows best practices in this case is that there is not really best practice for it (to the best of my knowledge). Currently, I'm also not quite sure that using `feat_dynamic_real` with unknown values in the future is useful. In a way, you make your problem **harder** because now you have to solve multiple forecast problems (the forecast of your target plus the forecast of your features) instead of **just** forecasting your target. 

I might be wrong here, of course :-).

**Reply by vikasmalhotra08** [reply]
Gotcha, thanks for the insights. It seems like I would have to go with either a Seq2Seq model or DeepVar model given that I have multiple time series which I know interact with each other and have an impact on the target that I am trying to forecast. 

Btw great library, I would love to contribute back someday. :)
